experimental/sql/src/projection_pushdown.py

```diff
@@ -0,0 +1,86 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from xdsl.ir import Operation, MLContext, Region, Block
+from typing import List, Type, Optional
+from xdsl.dialects.builtin import ArrayAttr, StringAttr, ModuleOp, IntegerAttr
+
+from xdsl.pattern_rewriter import (RewritePattern, GreedyRewritePatternApplier,
+                                   PatternRewriteWalker, PatternRewriter,
+                                   op_type_rewrite_pattern)
+
+import dialects.rel_alg as RelAlg
+
+
+@dataclass
+class ProjectionPushdownRewriter(RewritePattern):
+
+  def find_cols_in_expr(self, op: RelAlg.Expression) -> list[str]:
+    """
+    Returns the list of cols that are used in this operation or its children.
+    """
+    if isinstance(op, RelAlg.BinOp):
+      return self.find_cols_in_expr(op.lhs.op) + self.find_cols_in_expr(
+          op.rhs.op)
```

Comment:
You are right. i initially implemented it like this since the different child regions are named differently, but I can just go over the `op.regions` field.